rust/src/editor.rs

```diff
@@ -177,6 +178,29 @@ impl Editor {
         }
     }
 
+    fn cursor_start(&mut self) {
+        let start = self.view.sel_min() - self.col;
+        self.set_cursor(start, true);
+    }
+
+    fn cursor_end(&mut self) {
+        let current = self.view.sel_max();
+        let rope = self.text.clone();
+        let mut cursor = Cursor::new(&rope, current);
+        match cursor.next::<LinesMetric>() {
+            None => { self.set_cursor(current, true); },
+            Some(offset) => {
+                if cursor.is_boundary::<LinesMetric>() {
+                    if let Some(new) = rope.prev_grapheme_offset(offset) {
+                        self.set_cursor(new + 1, true);
```

Comment:
I thought so too, but `new` would advance to right before the last character. For example if I had the cursor before the `e` in `test` it would advance the cursor right before the last `t`. Is that desired behavior? I thought it would go to after the last `t`?